changeset:   63904:ca41c5663999
user:        Rafael ?vila de Esp?ndola <respindola@mozilla.com>
date:        Thu Mar 24 19:26:55 2011 -0400
summary:     Bug 623126 - Add constructor for nsDebugImpl, nsTraceRefcntImpl, EmptyEnumeratorImpl, and nsSimpleUnicharStreamFactory to placate CLang; r=bsmedberg

diff -r 92b43aa07b7d -r ca41c5663999 xpcom/base/nsDebugImpl.h
--- xpcom/base/nsDebugImpl.h	Thu Mar 24 16:45:07 2011 -0400
+++ xpcom/base/nsDebugImpl.h	Thu Mar 24 19:26:55 2011 -0400
@@ -40,6 +40,7 @@
 class nsDebugImpl : public nsIDebug2
 {
 public:
+    nsDebugImpl() {}
     NS_DECL_ISUPPORTS
     NS_DECL_NSIDEBUG
     NS_DECL_NSIDEBUG2
diff -r 92b43aa07b7d -r ca41c5663999 xpcom/base/nsTraceRefcntImpl.h
--- xpcom/base/nsTraceRefcntImpl.h	Thu Mar 24 16:45:07 2011 -0400
+++ xpcom/base/nsTraceRefcntImpl.h	Thu Mar 24 19:26:55 2011 -0400
@@ -44,6 +44,7 @@
 class nsTraceRefcntImpl : public nsITraceRefcnt
 {
 public:
+  nsTraceRefcntImpl() {}
   NS_DECL_ISUPPORTS
   NS_DECL_NSITRACEREFCNT
 
diff -r 92b43aa07b7d -r ca41c5663999 xpcom/glue/nsEnumeratorUtils.cpp
--- xpcom/glue/nsEnumeratorUtils.cpp	Thu Mar 24 16:45:07 2011 -0400
+++ xpcom/glue/nsEnumeratorUtils.cpp	Thu Mar 24 19:26:55 2011 -0400
@@ -52,6 +52,7 @@
                             public nsIStringEnumerator
 {
 public:
+    EmptyEnumeratorImpl() {}
     // nsISupports interface
     NS_DECL_ISUPPORTS_INHERITED  // not really inherited, but no mRefCnt
 
diff -r 92b43aa07b7d -r ca41c5663999 xpcom/io/nsUnicharInputStream.h
--- xpcom/io/nsUnicharInputStream.h	Thu Mar 24 16:45:07 2011 -0400
+++ xpcom/io/nsUnicharInputStream.h	Thu Mar 24 19:26:55 2011 -0400
@@ -51,6 +51,7 @@
   private nsISimpleUnicharStreamFactory
 {
 public:
+  nsSimpleUnicharStreamFactory() {}
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_NSIFACTORY
   NS_DECL_NSISIMPLEUNICHARSTREAMFACTORY
